src/functions/New-MockObject.ps1

```diff
@@ -29,9 +34,25 @@ https://pester.dev/docs/usage/mocking
     param (
         [Parameter(Mandatory = $true)]
         [ValidateNotNullOrEmpty()]
-        [type]$Type
+        [type]$Type,
+        [ValidateNotNullOrEmpty()]
+        [hashtable]$Properties
     )
 
-    [System.Runtime.Serialization.Formatterservices]::GetUninitializedObject($Type)
+    $mock = [System.Runtime.Serialization.Formatterservices]::GetUninitializedObject($Type)
+
+    if ($null -ne $Properties) {
+        foreach ($property in $Properties.GetEnumerator()) {
+            $addMemberSplat = @{
+                MemberType = [System.Management.Automation.PSMemberTypes]::NoteProperty
+                Name       = "$($property.Key)"
+                Value      = $property.Value
+                Force      = $true
+            }
+            $mock | Add-Member @addMemberSplat
+        }
+    }
```

Comment:
Not sure about the use of `Add-Member` as it completely replaces the property with breaking changes:
- Property -> NoteProperty membertype (not sure how big of an issue this is)
- Makes the new value writeable
- Doesn't enforce equal type
- Unlinks it from internal property which might be used in other ways